pkg/controller/network/securitygroup/controller.go

```diff
@@ -124,75 +145,104 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
+		GroupName:   aws.String(cr.Spec.ForProvider.GroupName),
+		VpcId:       cr.Spec.ForProvider.VPCID,
+		Description: aws.String(cr.Spec.ForProvider.Description),
+	}).Send(ctx)
 
-	rsp, err := req.Send(ctx)
 	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
 	}
 
-	// NOTE(muvaf): GroupID is used as external name instead of GroupName because
-	// there are cases where only GroupID is accepted as identifier.
-	meta.SetExternalName(cr, aws.StringValue(rsp.GroupId))
-	if err := e.kube.Update(ctx, cr); err != nil {
-		return managed.ExternalCreation{}, errors.Wrap(err, errPersistExternalName)
+	if result.CreateSecurityGroupOutput.GroupId == nil {
+		return managed.ExternalCreation{}, errors.New(errCreate)
 	}
 
-	cr.Status.SetConditions(runtimev1alpha1.Creating())
-	cr.UpdateExternalStatus(awsec2.SecurityGroup{GroupId: rsp.GroupId})
+	meta.SetExternalName(cr, aws.StringValue(result.GroupId))
 
-	return managed.ExternalCreation{}, nil
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
 }
 
-func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
-	cr, ok := mgd.(*v1alpha3.SecurityGroup)
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1beta1.SecurityGroup)
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
 	}
-	// TODO(soorena776): add more sophisticated Update logic, once we
-	// categorize immutable vs mutable fields (see #727)
 
-	if len(cr.Spec.Ingress) > 0 {
-		_, err := e.sg.AuthorizeSecurityGroupIngressRequest(&awsec2.AuthorizeSecurityGroupIngressInput{
+	response, err := e.sg.DescribeSecurityGroupsRequest(&awsec2.DescribeSecurityGroupsInput{
+		GroupIds: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(ec2.IsSecurityGroupNotFoundErr, err), errDescribe)
+	}
+
+	patch, err := ec2.CreateSGPatch(response.SecurityGroups[0], cr.Spec.ForProvider)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.New(errUpdate)
+	}
+
+	if len(patch.Tags) != 0 {
+		if _, err := e.sg.CreateTagsRequest(&awsec2.CreateTagsInput{
+			Resources: []string{meta.GetExternalName(cr)},
+			Tags:      v1beta1.GenerateEC2Tags(cr.Spec.ForProvider.Tags),
+		}).Send(ctx); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errCreateTags)
+		}
+	}
+
+	if patch.Ingress != nil {
+		if _, err := e.sg.AuthorizeSecurityGroupIngressRequest(&awsec2.AuthorizeSecurityGroupIngressInput{
 			GroupId:       aws.String(meta.GetExternalName(cr)),
-			IpPermissions: ec2.GenerateEC2Permissions(cr.Spec.Ingress),
-		}).Send(ctx)
-		if err != nil && !ec2.IsRuleAlreadyExistsErr(err) {
+			IpPermissions: v1beta1.BuildEC2Permissions(cr.Spec.ForProvider.Ingress),
+		}).Send(ctx); err != nil {
```

Comment:
It seems that previously we were ignoring errors that satisfied `ec2.IsRuleAlreadyExistsErr`. Do we no longer need to do that?

Reply:
Thanks for the catch!